docs/source/user_guide/kernel_programming/writing-range-kernel.rst

```diff
@@ -0,0 +1,142 @@
+A *range* kernel represents the simplest form of parallelism that can be
+expressed in numba-dpex using kapi. Such a kernel represents a data-parallel
+execution over a set of work items with each work item representing a logical
+thread of execution. :ref:`ex_vecadd_kernel` shows an example of a range kernel
+written in numba-dpex.
+
+.. code-block:: python
+    :linenos:
+    :caption: **Example:** Vector addition using a range kernel
+    :name: ex_vecadd_kernel
+    :emphasize-lines: 9,17
+
+    import dpnp
+    import numba_dpex.experimental as dpex
+    from numba_dpex import kernel_api as kapi
+
+
+    # Data parallel kernel implementing vector sum
+    @dpex.kernel
+    def vecadd(item: kapi.Item, a, b, c):
+        i = item.get_id(0)
+        c[i] = a[i] + b[i]
+
+
+    N = 1024
+    a = dpnp.ones(N)
+    b = dpnp.ones_like(a)
+    c = dpnp.zeros_like(a)
+    dpex.call_kernel(vecadd, kapi.Range(N), a, b, c)
+
+The highlighted lines in the example demonstrate the definition of the execution
+range on **line 17** and extraction of every work items' *id* or index position
+via the ``item.get_id`` call on **line 10**. An execution range comprising of
+1024 work items is defined when calling the kernel and each work item then
+executes a single addition. Note that the array sizes for the input and output
+arguments are equal to the size of the execution range. For very large arrays,
+the design will not scale as there is usually an upper bound for the range size
+depending on device. For most current Intel GPU devices, the maximum range size
+is 2^32 and a kernel requesting more work items than that bound will not
+execute. As such, programmers need to consider the size of the data and the
+access patterns for their kernels before scheduling a range kernel. The maximum
+number of work items can be queried programmatically as shown in
+:ref:`ex_max_work_item`.
```

Comment:
I think this chunk can be better explained in terms of maximum range limit instead of maximum array size because you have always have arrays larger than max range and map multiple array elements to a single work-item.
Moreover, this piece of text seems like a better fit in the below section where semantic rules are explained.

Reply:
I have restructured the section. Hopefully, it reads better now.